sdk/trace/provider.go

```diff
@@ -159,6 +161,10 @@ func (p *TracerProvider) Tracer(name string, opts ...trace.TracerOption) trace.T
 				provider:             p,
 				instrumentationScope: is,
 			}
+			env := os.Getenv("OTEL_GO_X_SELF_OBSERVABILITY")
```

Comment:
Can we add this as a feature in sdk/internal/x/x.go?  That is what i'm doing in https://github.com/open-telemetry/opentelemetry-go/pull/6393/files

Reply:
I did it initially but I found the package documentation less user-friendly.

Reply:
Sure. I will try to discuss it during the SIG meeting to discuss which pattern we prefer.

Reply:
Addressed https://github.com/open-telemetry/opentelemetry-go/pull/7027/commits/c40a561207cf9beec96234c8dfcfa2107dff4ae2